src/main/scala/Column.scala

```diff
@@ -1,22 +1,292 @@
+    inheritedPropertiesNode match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getTextDirection(inheritedProperties: Map[String, JsonNode]): String = {
+    val textDirectionNode = inheritedProperties.get("textDirection")
+    textDirectionNode match {
+      case Some(value) => value.asText()
+      case _           => "inherit"
+    }
+  }
+
+  def getSuppressOutput(columnProperties: Map[String, JsonNode]): Boolean = {
+    val suppressOutputNode = columnProperties.get("suppressOutput")
+    suppressOutputNode match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getVirtual(columnProperties: Map[String, JsonNode]): Boolean = {
+    val virtualNode = columnProperties.get("virtual")
+    virtualNode match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getRequired(inheritedProperties: Map[String, JsonNode]): Boolean = {
+    inheritedProperties.get("required") match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getDefault(inheritedProperties: Map[String, JsonNode]): String = {
+    inheritedProperties.get("default") match {
+      case Some(value) => value.asInstanceOf[TextNode].asText()
+      case _           => ""
+    }
+  }
+
+  def getId(columnProperties: Map[String, JsonNode]): Option[String] = {
+    val idNode = columnProperties.get("@id")
+    if (idNode.isDefined) Some(idNode.get.asText()) else None
+  }
+
+  def getName(
+      columnProperties: Map[String, JsonNode],
+      lang: String
+  ): Option[String] = {
+    val name = columnProperties.get("name")
+    val titles = columnProperties.get("titles")
+//
+//    name
+//      .map(n => n.asText())
+//      .orElse(
+//        titles.map(t => {
+//          val langNode = t.path(lang)
+//          if (langNode.isMissingNode) {
+//            langNode.elements.asScalaArray(0).asText()
+//          }
+//        })
+//      )
+
+    if (name.isDefined) {
+      Some(name.get.asInstanceOf[TextNode].asText())
+    } else if (titles.isDefined && titles.get.path(lang).isMissingNode) {
+      val langArray = Array.from(
+        titles.get.path(lang).elements().asScala
+      )
+      if (langArray.nonEmpty) {
+        Some(langArray(0).asText())
+      } else None
+    } else None // Not sure what to return here. Hope it does not reach here
+  }
+
+  def getNullParam(
+      inheritedProperties: Map[String, JsonNode]
+  ): Array[String] = {
+    inheritedProperties.get("null") match {
+      case Some(value) => {
+        value match {
+          case a: ArrayNode => {
+            var nullParamsToReturn = Array[String]()
+            val nullParams = Array.from(a.elements.asScala)
+            for (np <- nullParams)
+              nullParamsToReturn :+= np.asText()
+            nullParamsToReturn
+          }
+          case s: TextNode => Array[String](s.asText())
```

Comment:
What about if it's not an array or text node? Presumably we want a warning? We certainly want to return *something*

Reply:
I think instead of returning a warning, it's better to throw an exception here. As null according to the documentation is,
 `null — the string or strings which cause the value of cells having string value matching any of these values to be null.`
 
 Even if we set warning and return what we got here (may be an object node) we might not be able to deal with it later.

Reply:
Great, it's good that you've checked the CSV-W spec to figure out what should happen here. 

It sounds like throwing an exception is the right thing to do.

Reply:
I have changed the code to throw an exception if its not an array or text-node.